cdqa/utils/converters.py

```diff
@@ -135,12 +135,39 @@ def pdf_converter(directory_path):
             df.loc[i] = [pdf, None]
             raw = parser.from_file(os.path.join(directory_path, pdf))
             s = raw["content"]
-            paragraphs = re.split(u"\n(?=\u2028|[A-Z-0-9])", s)
+            paragraphs = re.split("\n(?=\u2028|[A-Z-0-9])", s)
             list_par = []
-            for p in paragraphs:
-                if len(p) >= 200:
-                    list_par.append(p.replace("\n", ""))
-                df.loc[i, "paragraphs"] = list_par
+            line_para = ""
+            if line_paragraph:
+                for p in paragraphs:
+                    if len(p) >= 200:
+                        if line_para:
+                            list_par.append(line_para)
+                            line_para = ""
+                            list_par.append(p.replace("\n", ""))
+                        else:
+                            list_par.append(p.replace("\n", ""))
+                    else:
+                        line = p
+                        if not line.isspace():
+                            line = line.replace("\n", "").strip()
+                            line_para = line_para.replace("\n", "").strip()
+                            line_para = line_para + f" {line}"
+                            continue
+                        else:
+                            continue
+                    df.loc[i, "paragraphs"] = list_par
+                else:
+                    if line_para:
+                        line_para = line_para.replace("\n", "").strip()
+                        list_par.append(line_para)
+                    df.loc[i, "paragraphs"] = list_par
+            else:
+                for p in paragraphs:
+                    if len(p) >= 200:
+                        list_par.append(p.replace("\n", ""))
+                    df.loc[i, "paragraphs"] = list_par
```

Comment:
Overall, I think this code is pretty messy, I spent some minutes trying to understand it and it's a little bit painful... Isn't there a way to clean it a bit?
Also, you can add some comments to help others (and me) understand it, to make changes in the future.

Reply:
Made the code better and added docstring (I may have taken it a bit too far :P). Let me know what you think of it and of the further changes if needed.